### DISCUSSION TITLE
Custom secret engine

### DISCUSSION THREAD

**Discussion by teneko** [rajanadar/VaultSharp#263] [Q&A]
I am very irritated. It is not possible to implement a custom secret engine from a custom plugin I implemented myself (`vault secrets enable -path=<plugin-name> -plugin-name=<plugin-name> plugin`)? What I see is that all secret engines are using an internal Polymath instance (the name itself is very irritating) and that instance is capable of making "native" vault calls.

Is there no way to implement a custom secret engine?

**Comment by rajanadar** [comment]
hi @teneko 

Hope you got help with your irritation over the last few months.
Can you please expand your use case in a GH issue?
